### ISSUE TITLE
[CT-3045] Add documentation of `meta` as supported attribute of `SemanticModel` nodes

### ISSUE THREAD

**Issue by QMalcolm** [dbt-labs/dbt-core#8514]
After the completion of #8511 we should update the [meta documentation page](https://docs.getdbt.com/reference/resource-configs/meta) to note that `meta` is also a supported field of `SemanticModel`s

**Comment by QMalcolm** [comment]
Blocked by https://github.com/dbt-labs/dbt-core/issues/8511

**Comment by jtcohen6** [comment]
Closing in favor of https://github.com/dbt-labs/docs.getdbt.com/issues/4180
